usaspending_api/etl/es_etl_helpers.py

```diff
@@ -0,0 +1,465 @@
+            sleep(15)
+
+    printf({'msg': 'Completed Elasticsearch data load', 'f': 'ES Ingest'})
+    return
+
+
+def streaming_post_to_es(client, chunk, index_name, job_id=None):
+    success, failed = 0, 0
+    try:
+        for ok, item in helpers.streaming_bulk(client, chunk, index=index_name, doc_type='transaction_mapping'):
+            success = [success, success + 1][ok]
+            failed = [failed + 1, failed][ok]
+
+    except Exception as e:
+        print('MASSIVE FAIL!!!\n\n{}\n\n{}'.format(str(e)[:5000], '*' * 80))
+        raise SystemExit
+
+    printf({'msg': 'Success: {}, Fails: {}'.format(success, failed), 'job': job_id, 'f': 'ES Ingest'})
+    return success, failed
+
+
+def post_to_elasticsearch(client, job, config, chunksize=250000):
+    printf({'msg': 'Populating ES Index "{}"'.format(job.index), 'job': job.name, 'f': 'ES Ingest'})
+    start = perf_counter()
+    try:
+        does_index_exist = client.indices.exists(job.index)
+    except Exception as e:
+        print(e)
+        raise SystemExit
+    if not does_index_exist:
+        printf({'msg': 'Creating index "{}"'.format(job.index), 'job': job.name, 'f': 'ES Ingest'})
+        client.indices.create(index=job.index)  # removed body paramter since behavior wasn't reliable
+    elif does_index_exist and config['recreate']:
+        printf({'msg': 'Deleting existing index "{}"'.format(job.index), 'job': job.name, 'f': 'ES Ingest'})
+        client.indices.delete(job.index)
+
+    csv_generator = csv_chunk_gen(job.csv, chunksize, job.name)
+    for count, chunk in enumerate(csv_generator):
+        if len(chunk) == 0:
+            printf({'msg': 'No documents to add/delete for chunk #{}'.format(count), 'f': 'ES Ingest', 'job': job.name})
+            continue
+        iteration = perf_counter()
+        if config['recreate'] is False:
+            id_list = [{'key': c[UNIVERSAL_TRANSACTION_ID_NAME], 'col': UNIVERSAL_TRANSACTION_ID_NAME} for c in chunk]
+            delete_transactions_from_es(client, id_list, job.name, config, job.index)
+
+        current_rows = '({}-{})'.format(count * chunksize + 1, count * chunksize + len(chunk))
+        printf({
+            'msg': 'Streaming to ES #{} rows [{}/{}]'.format(count, current_rows, job.count),
+            'job': job.name,
+            'f': 'ES Ingest'
+        })
+        streaming_post_to_es(client, chunk, job.index, job.name)
+        printf({
+            'msg': 'Iteration group #{} took {}s'.format(count, perf_counter() - iteration),
+            'job': job.name,
+            'f': 'ES Ingest'
+        })
+    printf({
+        'msg': 'Elasticsearch Index loading took {}s'.format(perf_counter() - start),
+        'job': job.name,
+        'f': 'ES Ingest'
+    })
+
+
+def deleted_transactions(client, config):
+    deleted_ids = gather_deleted_ids(config)
+    id_list = [{'key': c, 'col': UNIVERSAL_TRANSACTION_ID_NAME} for c in deleted_ids]
```

Comment:
Can you rename `c` to `deleted_id` here for readability?

Reply:
yes